sotodlib/tod_ops/binning.py

```diff
@@ -38,27 +42,33 @@ def bin_signal(aman, bin_by, signal=None,
         signal = aman.signal
     if range is None:
         range = (np.nanmin(bin_by), np.nanmax(bin_by))
+    
+    if weight_for_signal is None:
+        weight_for_signal = np.ones(aman.samps.count)
         
-    # bin `bin_by` data
-    bin_counts, bin_edges = np.histogram(bin_by, bins=bins, range=range)
+    # get bin_edges
+    _, bin_edges = np.histogram(bin_by, bins=bins, range=range,)
     bin_centers = (bin_edges[1] - bin_edges[0])/2. + bin_edges[:-1] # edge to center
     nbins = len(bin_centers)
     
     # prepare binned signal array
     binned_signal = np.full([aman.dets.count, nbins], np.nan)
     binned_signal_squared_mean = np.full([aman.dets.count, nbins], np.nan)
     binned_signal_sigma = np.full([aman.dets.count, nbins], np.nan)
-
+        
     # bin tod
     if flags is None:
+        bin_counts, _ = np.histogram(bin_by, bins=bins, range=range, weights = weight_for_signal)
```

Comment:
Can we return `counts` in addition to `binned_signal_sigma` and also get that wrapped into the binning dict in the azss + hwpss binning fn's?